lib/utils/configurationError.js

```diff
@@ -1,13 +1,14 @@
 'use strict';
 
+/** @typedef {import('stylelint').ConfigurationError} ConfigurationError */
+
 /**
  * Create configurationError from text and set CLI exit code
  * @param {string} text
- * @returns {Object}
+ * @returns {ConfigurationError}
  */
-module.exports = function (text) /* Object */ {
-	/** @type {Error & {code?: number}} */
-	const err = new Error(text);
+module.exports = function (text) {
+	const err = /** @type {ConfigurationError} */ (new Error(text));
```

Comment:
[trivial] IMO, separating the lines is a bit readable. But you can keep it as-is since it is just my preference. 😃 

```suggestion
	/** @type {ConfigurationError} */
	const err = new Error(text);
```

Reply:
Placing the comment behind the declaration is equivalent to:

```ts
const err: ConfigurationError = new Error(text);
```

This won't work because `code` isn't defined on `Error`. Hence why I changed it to what it is now, which is the equivalent of the typecast:

```ts
const err = new Error(text) as ConfigurationError;
```

Reply:
Ah, I see why. Thanks for the help! 😊